src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -305,6 +311,23 @@ private object ConvertFromStorageFormat(byte[] contents, Type stateType)
 
             return result;
         }
+
+        public async IAsyncEnumerable<StorageEntry> GetAll([EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            var regex = new Regex("(?<name>[^-]+)-(?<reference>[^-]+).json");
+            await foreach (var item in this.container.GetBlobsAsync(cancellationToken: cancellationToken))
```

Comment:
We probably need a way to store progress periodically so that we make forward progress after a failure. Eg, store the last migrated key periodically and scan blob names starting from that id on recovery.

Reply:
It may also be advantageous to support a graceful "pausing" of the migration. Some environments may only want to run this during periods of lower systems load.